### DISCUSSION TITLE
Unreferenced columns definition

### DISCUSSION THREAD

**Discussion by hassan75016** [sql-bi/Bravo#417] [Q&A] [answered]
Hello everyone - Just ran Bravo on one of my PBIX containing a star schema. The results show a lot of "unreferenced" columns. These are the ones which are not linked to the fact/dimension tables. Can you please explain the concept of "unreferenced" columns? A lot of them are being utilized inside the report however, they are marked as unreferenced.

**Comment by albertospelta** [comment, marked as answer]
@hassan75016 "unreferenced" columns are those columns that are not used in any measure or in any relationship within the model. In the current version Bravo can not determine the column usage at report/visual level so, before removing any "unreferenced" column, make sure you are not using these columns in any reports. 

https://www.youtube.com/watch?v=Uq4ivJwpnoc&t=182s

We are evaluating a technique to detect columns referenced in visuals of the same PBIX and we have an open issue here #212

**Reply by hassan75016** [reply]
@albertospelta many thanks for the YT link and clarification. Hope we get the feature currently under evaluation for detecting columns referenced in visuals. Good luck.
